lib/applicationModel.js

```diff
@@ -0,0 +1,135 @@
+const ModelClass = require('./modelClass.js');
+const debugApplicationModel = require('debug')('applicationModel');
+const instanceMap = new Map();
+
+class ApplicationModel {
+  constructor(caller) {
+    this.caller = caller;
+    debugApplicationModel(`constructor for ${caller} ++++++++++++++++++++++++++++++++++++++++++`);
+    instanceMap.set(caller, this);
+    debugApplicationModel(instanceMap);
+  }
+
+  getModelClass(schemaName) {
+    debugApplicationModel(`getModelClass for caller ${this.caller} schema ${schemaName}`);
+    this.setupSuperClassMap();
+    this.setupModelClassMap();
+    const modelClass = this.modelClassMap[schemaName];
+    debugApplicationModel(`returning modelClass for  caller ${this.caller} ${schemaName}`);
+    debugApplicationModel(modelClass);
+    return modelClass;
+  };
+
+  getSchema(schemaName) {
+    //debugApplicationModel(ApplicationModel.asyncapi.allSchemas());
+    const schema = ApplicationModel.asyncapi.components().schema(schemaName);
+    //debugApplicationModel(`getSchema for ${schemaName}`);
+    //debugApplicationModel(schema);
+    return schema;
+  }
+
+  setupSuperClassMap() {
+    if (!this.superClassMap) {
+      this.superClassMap = new Map();
+      this.anonymousSchemaToSubClassMap = new Map();
+      debugApplicationModel('-------- SCHEMAS -------------');
+      debugApplicationModel(ApplicationModel.asyncapi.allSchemas());
+      ApplicationModel.asyncapi.allSchemas().forEach((schema, schemaName) => {
+        debugApplicationModel(`${schemaName}:`);
+        debugApplicationModel(schema);
+        const allOf = schema.allOf();
+        if (allOf) {
```

Comment:
What's the plan for related keywords like anyOf, oneOf, and not? Something to implement later? I dont wanna scope creep but it would seem odd as a user.

Reply:
I'll address those as people request them.